libnavigation-core/src/main/java/com/mapbox/navigation/core/formatter/MapboxDistanceUtil.kt

```diff
@@ -37,31 +39,35 @@ object MapboxDistanceUtil {
         context: Context,
         locale: Locale
     ): FormattedDistanceData {
-        return when (distanceInMeters) {
-            !in 0.0..Double.MAX_VALUE -> {
-                formatDistanceAndSuffixForSmallUnit(
-                    0.0,
-                    roundingIncrement,
-                    unitType,
-                    context,
-                    locale
-                )
-            }
-            in 0.0..smallDistanceUpperThresholdInMeters -> {
-                formatDistanceAndSuffixForSmallUnit(
-                    distanceInMeters,
-                    roundingIncrement,
-                    unitType,
-                    context,
-                    locale
-                )
-            }
-            in smallDistanceUpperThresholdInMeters..mediumDistanceUpperThresholdInMeters -> {
-                formatDistanceAndSuffixForLargeUnit(distanceInMeters, 1, unitType, context, locale)
-            }
-            else -> {
-                formatDistanceAndSuffixForLargeUnit(distanceInMeters, 0, unitType, context, locale)
-            }
+        return when (getFormattingRange(distanceInMeters, unitType)) {
+            FormattingRange.INVALID -> formatDistanceAndSuffixForSmallUnit(
+                distanceInMeters,
+                roundingIncrement,
```

Comment:
It was 0 before but in that case roundingIncrement was applied to an artificial zero distance (when the original one was negative). So when invoking 
```
formatDistance(
        distanceInMeters = -234.4,
        roundingIncrement = RoundingIncrement.INCREMENT_FIFTY,
        unitType = UnitType.METRIC,
        context
    )
```
we used to get 50 meters, not 0 meters. I changed that: now it's 0 meters.

Reply:
OK, I reverted it because there was a test for this behaviour.